File: app/src/main/java/com/chris/eban/AndroidComponentModule.java
package com.chris.eban;

import com.chris.eban.presenter.EventDetailActivity;
import com.chris.eban.presenter.MainActivity;
import com.chris.eban.presenter.event.EventListFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@SuppressWarnings("unused")
@Module
abstract class AndroidComponentModule {

    @ContributesAndroidInjector
    public abstract MainActivity mainActivity();

    @ContributesAndroidInjector
    public abstract EventDetailActivity createEventActivity();


    //----------------  fragment   -----------------

//    @ContributesAndroidInjector
//    public abstract MainFragment mainFragment();

    @ContributesAndroidInjector
    public abstract EventListFragment eventListFragment();
}

File: app/src/main/java/com/chris/eban/AndroidComponentModule.kt
package com.chris.eban

import com.chris.eban.presenter.MainActivity
import com.chris.eban.presenter.ManagerFragment
import com.chris.eban.presenter.event.EventDetailActivity
import com.chris.eban.presenter.event.EventListFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
internal abstract class AndroidComponentModule {

    @ContributesAndroidInjector
    abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector
    abstract fun createEventActivity(): EventDetailActivity


    //----------------  fragment   -----------------

    //    @ContributesAndroidInjector
    //    public abstract MainFragment mainFragment();

    @ContributesAndroidInjector
    abstract fun eventListFragment(): EventListFragment

    @ContributesAndroidInjector
    abstract fun managerFragment(): ManagerFragment
}
